### ISSUE TITLE
100% CPU usage by language server

### ISSUE THREAD

**Issue by pzmarzly** [Eugleo/magic-racket#12]
The language server does not work for me, and it uses all CPU time on a core it runs on.

To reproduce:
- create new file `anything.rkt`
- open it
- hear your CPU fan spinning

I had to downgrade with:

```text
code --install-extension evzen-wybitul.magic-racket@0.4.5
```

Ubuntu 19.10, Racket 7.6 (installed from [official PPA](http://ppa.launchpad.net/plt/racket/ubuntu)), VS Codium 1.44.2 (ff915844119ce9485abfe8aa9076ec76b5300ddd).

**Comment by pzmarzly** [comment]
(I forgot to mention I was using Live Share all the time, but I was a host so it shouldn't matter).

**Comment by Eugleo** [comment]
Thanks for reporting, I’ll look into it.

**Comment by Eugleo** [comment]
Weird, I'm not able to reproduce this. I'm on Racket 7.6 installed through homebrew (MacOS). Do you see any other errors, like "Unable to launch the language server" or similar?

Also, could you please run `raco install racket-langserver`, try again and report the results? Thanks.

**Comment by pzmarzly** [comment]
The last thing I see in Extension Host logs is pretty standard:

```text
[info] ExtensionService#_doActivateExtension evzen-wybitul.magic-racket {"startup":false,"extensionId":{"value":"evzen-wybitul.magic-racket","_lower":"evzen-wybitul.magic-racket"},"activationEvent":"onLanguage:racket"}
[2020-05-01 19:25:01.151] [exthost] [info] ExtensionService#loadCommonJSModule file:///home/pzmarzly/.vscode-oss/extensions/evzen-wybitul.magic-racket-0.5.0/out/extension.js
```

The process being stuck is `racket /home/pzmarzly/.vscode-oss/extensions/evzen-wybitul.magic-racket-0.5.0/racket-langserver/main.rkt`. So I think it may be that the issue is with the language server itself. But I don't know how to test language server from the terminal, or how to observe traffic. Interestingly, the problem does not occur for non-empty files, so a working workaround is to create new files via `echo "#lang racket" > file.rkt`. Even in empty files, the issue does not occur every time, but a sure way to trigger it for me is to hover the mouse mouse over the text cursor (when it's in row 1 col 1 of empty file), it will open a pop-up with message "Loading..." and the language server will crash.

OTOH I downloaded this extension's repo, and I'm getting another kind of error there, which I think is unrelated:

```
hash-ref: no value found for key
  key: 'file:///home/pzmarzly/magic-racket/racket-langserver/methods.rkt
  context...:
   /home/pzmarzly/.vscode-oss/extensions/evzen-wybitul.magic-racket-0.5.0/racket-langserver/text-document.rkt:210:0: get-doc-refs
   /home/pzmarzly/.vscode-oss/extensions/evzen-wybitul.magic-racket-0.5.0/racket-langserver/text-document.rkt:194:0: document-highlight
   /usr/share/racket/collects/racket/contract/private/arrow-val-first.rkt:486:18
   /home/pzmarzly/.vscode-oss/extensions/evzen-wybitul.magic-racket-0.5.0/racket-langserver/methods.rkt:26:0: process-message
   /usr/share/racket/collects/racket/contract/private/arrow-val-first.rkt:486:18
   /home/pzmarzly/.vscode-oss/extensions/evzen-wybitul.magic-racket-0.5.0/racket-langserver/main.rkt:54:2: consume

[Error - 8:21:14 PM] Request textDocument/hover failed.
  Message: internal error in method "textDocument/hover"
  Code: -32603 
[Error - 8:21:14 PM] Request textDocument/documentHighlight failed.
  Message: internal error in method "textDocument/documentHighlight"
  Code: -32603 
```

I ran `raco pkg install racket-langserver`, it didn't change anything.

**Comment by Eugleo** [comment]
In a few seconds there will be a version v0.5.2, that will include a setting called `magic-racket > trace > server`. Set that from `off` to `verbose` and then try to reproduce the error you're seeing, on some minimal example. 

In the output window, in the channel called `Racket language client`, there will be some logs about the communication between the client and the server. Copy it into some file and send it to me attached to an issue reply, please.

You can also see how the logs look like [here](https://microsoft.github.io/language-server-protocol/inspector/) compared to mine: 

<img width="828" alt="Snímek obrazovky 2020-05-01 v 21 12 39" src="https://user-images.githubusercontent.com/21257632/80834192-8059c480-8bf0-11ea-9cd7-839973d401aa.png">

**Comment by pzmarzly** [comment]
Log when it hangs on open: [log.txt](https://github.com/Eugleo/magic-racket/files/4566007/log.txt)

Log where I also try hovering: 
[log-hover.txt](https://github.com/Eugleo/magic-racket/files/4566011/log-hover.txt)

**Comment by Eugleo** [comment]
Thanks, it's pleasure to be debugging with you! I think I found the cause and will look into it tomorrow. If you're interested in what's been happening, read on...

It's the `textDocument/documentSymbol` request, a.k.a. the thing that shows the file's symbols in the Outline window. As I don't have the outline window open by default — and you do, it seems — VS Code lazily doesn't request the symbol information, and thus doesn't trigger the bug you were experiencing. But, thanks to the logs, I was able to figure this out.
